pandas/core/indexing.py

```diff
@@ -1850,7 +1850,8 @@ def _setitem_single_block(self, indexer, value, name: str):
                     for i, idx in enumerate(indexer)
                     if i != info_axis
                 )
-                and item_labels.is_unique
+                and len(item_labels.get_indexer_for([item_labels[indexer[info_axis]]]))
+                == 1
```

Comment:
lets separate this out into a nested condition and define `item_labels[indexer[info_axis]]` once, then reuse it on L1856

Reply:
Done